src/extensions/extensionsHelper.ts

```diff
@@ -503,52 +531,284 @@ export async function publishExtensionVersionFromLocalSource(args: {
         )
     );
   }
-  displayReleaseNotes(args.publisherId, args.extensionId, extensionSpec.version, notes);
-  if (
-    !(await confirm({
+
+  if (args.latestVersion) {
+    if (semver.lt(extensionSpec.version, args.latestVersion)) {
+      throw new FirebaseError(
+        `The version you are trying to publish (${clc.bold(
+          extensionSpec.version
+        )}) is lower than the current version (${clc.bold(
+          args.latestVersion
+        )}) for the extension '${clc.bold(
+          extensionRef
+        )}'. Please make sure this version is greater than the current version (${clc.bold(
+          args.latestVersion
+        )}) inside of extension.yaml.\n`,
+        { exit: 104 }
+      );
+    } else if (semver.eq(extensionSpec.version, args.latestVersion)) {
+      throw new FirebaseError(
+        `The version you are trying to publish (${clc.bold(
+          extensionSpec.version
+        )}) already exists for Extension '${clc.bold(
+          extensionRef
+        )}'. Please increment the version inside of extension.yaml.\n`,
+        { exit: 103 }
+      );
+    }
+  }
+
+  return { extensionSpec, notes };
+}
+
+/**
+ * Publishes an Extension version from a remote repo.
+ *
+ * @param publisherId the ID of the Publisher this Extension will be published under
+ * @param extensionId the ID of the Extension to be published
+ * @param repoUri the URI of the repo where this Extension's source exists
+ * @param sourceRef the commit hash, branch, or tag name in the repo to publish from
+ * @param extensionRoot the root directory that contains this Extension's source
+ * @param stage the release stage to publish
+ * @param nonInteractive whether to display prompts
+ * @param force whether to force confirmations
+ */
+export async function publishExtensionVersionFromRemoteRepo(args: {
+  publisherId: string;
+  extensionId: string;
+  repoUri: string;
+  sourceRef: string;
+  extensionRoot: string;
+  stage: ReleaseStage;
+  nonInteractive: boolean;
+  force: boolean;
+}): Promise<ExtensionVersion | undefined> {
+  const extensionRef = `${args.publisherId}/${args.extensionId}`;
+
+  // Check if Extension already exists.
+  let extension: Extension | undefined;
+  try {
+    extension = await getExtension(extensionRef);
+  } catch (err: any) {
+    // Silently fail and continue the publish flow if Extension not found.
+  }
+
+  // Prompt for repo URI and validate that it hasn't changed if previously set.
+  if (args.repoUri && !repoRegex.test(args.repoUri)) {
+    throw new FirebaseError("Repo URI follows this format: https://github.com/<user>/<repo>");
+  }
+  let repoUri = args.repoUri || extension?.repoUri;
+  if (!repoUri) {
+    repoUri = await promptForValidRepoURI();
+  }
+  if (extension?.repoUri) {
```

Comment:
Optional: Consider unnesting this if statement, its a little hard to follow as is. Something like:
```
if (extension?.repoUri && extension.repoUri !== repoUri) {
  throw ...
}

if (!extension?.repoUri) {
  logger.info("You are about to associate...")
  const confirmed = await confirm(...)
  if (!confirmed) return
}
```